src/drops.cpp

```diff
@@ -10,114 +12,54 @@
+   // ignore transfers sent from this contract
+   if (from == "eosio.ram"_n || to != get_self() || from == get_self()) {
+      return 0;
+   }
 
    check(get_first_receiver() == "eosio.token"_n, "Only the eosio.token contract may send tokens to this contract.");
-   check(quantity.amount > 0, "The transaction amount must be a positive value.");
    check(quantity.symbol == EOS, "Only the system token is accepted for transfers.");
    check(!memo.empty(), ERROR_INVALID_MEMO);
-
-   // Process the memo field to determine the number of drops to generate
-   const vector<string> parsed = split(memo, ',');
-   if (parsed[0] == "unbind") {
-      check(parsed.size() == 1, "Memo data must only contain 1 value of 'unbind'.");
-      return do_unbind(from, quantity);
-   } else {
-      check(parsed.size() == 2, ERROR_INVALID_MEMO);
-
-      const int64_t amount = to_number(parsed[0]);
-      check(amount > 0, "The drops amount must be a positive value.");
-      const string data = parsed[1];
-      return do_generate(from, quantity, amount, data);
-   }
-}
-
-drops::generate_return_value
-drops::do_generate(const name account, const asset tokens_received, const uint64_t drop_quantity, const string data)
-{
-   check_is_enabled();
-
-   // Buy the required RAM for the bound drops
-   const asset ram_purchase_cost = buy_required_ram(drop_quantity, tokens_received);
-
-   // Iterate over all drops to be created and insert them into the drop table
-   emplace_drops(get_self(), account, data, drop_quantity);
-
-   // Refund any remaining tokens to the sender
-   asset refunded = refund_remaining_tokens(account, tokens_received, ram_purchase_cost);
-
-   return {
-      (uint32_t)drop_quantity, // drops bought
-      ram_purchase_cost,       // cost
-      refunded,                // refund
-      drop_quantity,           // total drops
-   };
-}
-
-drops::generate_return_value drops::do_unbind(const name account, const asset tokens_received)
-{
-   check_is_enabled();
-
-   // Find the unbind request of the owner
-   unbind_table           unbinds(get_self(), get_self().value);
-   auto&                  unbind    = unbinds.get(account.value, "No unbind request found for account.");
-   const vector<uint64_t> drops_ids = unbind.drops_ids;
-
-   // Buy the required RAM for the bound drops
-   const asset ram_purchase_cost = buy_required_ram(drops_ids.size(), tokens_received);
-
-   // Recreate all selected drops with new bound value (false)
-   for (const uint64_t drop_id : drops_ids) {
-      modify_ram_payer(drop_id, account, get_self());
+   check_is_enabled(get_self());
+
+   // Process the memo field by comma delimiting the string
+   // Memo payload must contain to 2 fields
+   const vector<string> parsed = utils::split(memo, ',');
+   check(parsed.size() >= 2, ERROR_INVALID_MEMO);
+   const string operation = parsed[0];
+
+   // Buy RAM: "buyram,<receiver>"
+   // If no receiver, the RAM would be credited to sender
+   if (operation == "buyram") {
```

Comment:
If we keep this `if` statement, we need an `else` that throws an exception. 

Right now if you use any string other than `buyram`, it just accepts the tokens and doesn't do anything else.

If we're just simplifying this down to the memo = account name, then I don't think this is an issue. Just figured I'd mention it though.

Reply:
Yep, resolved with new memo schema 👍 

However, for that example, it was still failing since there was a `check(false)` as fallback